### ISSUE TITLE
User can't sort table by sort icon on `DataTable` - keyboard accessibility

### ISSUE THREAD

**Issue by spli02** [fortanix/baklava#377]
User can't sort table by the sort icon on `DataTable` when the page is controlled by keyboard as the sort icon is not focusable.

https://github.com/user-attachments/assets/d2d26004-1460-43a6-bae0-2e02511db9bd
